src/prompt/variable.rs

```diff
@@ -134,10 +398,41 @@ pub struct Decimal;
 
 impl VariableInput for Decimal {
     fn type_name(&self) -> &str { "decimal" }
-    fn parse(&self, input: &str) -> Result<Value, Error> {
-        let dec: BigDecimal = input.parse().map_err(no_pos_err)?;
-        let dec = dec.try_into().map_err(no_pos_err)?;
-        Ok(Value::Decimal(dec))
+    fn parse<'a>(&self, input: &'a str, _flags: InputFlags) -> ParseResult<'a> {
+        context(
+            "decimal",
+            map(
+                map_res(
+                    map_res(
+                        recognize_float_parts,
```

Comment:
I think that if you use `recognize` on `recognize_float_parts`, you can get back the string and then parse that directly.

https://docs.rs/nom/latest/nom/combinator/fn.recognize.html